pkg/cmd/gpg-key/add/http.go

```diff
@@ -0,0 +1,67 @@
+package add
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"io"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/cli/cli/v2/api"
+	"github.com/cli/cli/v2/internal/ghinstance"
+)
+
+var scopesError = errors.New("insufficient OAuth scopes")
+
+func GPGKeyUpload(httpClient *http.Client, hostname string, keyFile io.Reader) error {
```

Comment:
Nitpick: it doesn't seem it's necessary that this function is exported as public
```suggestion
func gpgKeyUpload(httpClient *http.Client, hostname string, keyFile io.Reader) error {
```